### PR TITLE
Add ability to specify export options

### PR DESCRIPTION
**Pull request by jvinod**
**What this PR does / why we need it**:
Add ability to specify export options. Specifically to be used for SELinux support.

**Special notes for your reviewer**:
Needs to support update via SDK.

### CODE REVIEW

File: api/api.proto [hunk 1]
```diff
@@ -281,6 +281,28 @@ message IoStrategy {
...
+  // Custom the volume is exported over custom interace.
+  CUSTOM = 4;
+}
+
+// ExportSpec defines how the volume is exported..
+message ExportSpec {
+   // ExportProtocol defines how the volume is exported.
+   ExportProtocol export_protocol = 1;
+   // ExportOptions options exporting the volume.
+   string export_options = 2;
```
Comment (adityadani) [inline api/api.proto]: Are these comma separated options ? Why not use map?

File: api/spec/spec_handler.go [hunk 2]
```diff
@@ -366,6 +366,26 @@ func (d *specHandler) UpdateSpecFromOpts(opts map[string]string, spec *api.Volum
...
+			if v == api.SpecExportProtocolPXD {
+				spec.ExportSpec.ExportProtocol = api.ExportProtocol_PXD
+			} else if v == api.SpecExportProtocolNFS {
+				spec.ExportSpec.ExportProtocol = api.ExportProtocol_NFS
+			} else if v == api.SpecExportProtocolISCSI {
+				spec.ExportSpec.ExportProtocol = api.ExportProtocol_ISCSI
+			} else if v == api.SpecExportProtocolCustom {
+				spec.ExportSpec.ExportProtocol = api.ExportProtocol_CUSTOM
+			} else {
+				return nil, nil, nil, fmt.Errorf("Unsupported Export Protocol %v", v)
```
Comment (adityadani) [inline api/spec/spec_handler.go]: nit: errors start with lowercase letters.

File: api/api.proto (hunk 1 above)
Comment (adityadani) [reply api/api.proto]: What are some examples of options?

Comment (jvinod) [reply api/api.proto]: The export string will be appended as-is to the default export string e..g "security_label"

**Comment by jvinod** [comment]
closing in favor of https://github.com/libopenstorage/openstorage/pull/1303

**Pull request closed without merging**